js-html-games/bogscript.js

```diff
@@ -1,7 +1,378 @@
+                row1.appendChild(col1);
+            }
+        }
+        createCols();
+        // const allDivs = document.querySelectorAll(".letter");
+        // for (let i = 0; i < allDivs.length; i++) {
+        //     allDivs[i].innerText = i + 1;
+        // }
+    }
+}
+
+function createKeyboard(rightWord, wordList) {
+    const enter = document.createElement('button');
+    enter.setAttribute('class', "enter");
+    enter.innerText = "Enter";
+    enter.addEventListener('click', function () {
+        matchWord(rightWord, typedInput.join(""), wordList)
+        typedInput = [];
+        console.log(typedInput.join(""));
+    })
+    const back = document.createElement('button');
+    back.setAttribute('class', "back");
+    back.innerText = "Backspace";
+    back.addEventListener('click', function () {
+        erase();
+        typedInput.pop();
+    })
+    for (let i = 0; i < keyboardLetters.length; i++) {
+        const alphaLetter = document.createElement('button');
+        alphaLetter.setAttribute('class', "alphaLetter");
+        alphaLetter.innerText = keyboardLetters[i];
+        alphaLetter.addEventListener('click', function () {
+            typedInput.push(keyboardLetters[i]);
+            type();
+            console.log(typedInput);
+            console.log(keyboardLetters[i]);
+        })
+        if (i < 10) {
+            document.getElementById("one").appendChild(alphaLetter);
+        }
+        else if (i > 18) {
+            document.getElementById("three").appendChild(alphaLetter);
+        }
+        else {
+            document.getElementById("two").appendChild(alphaLetter);
+        }
+    }
+    document.getElementById("three").appendChild(enter);
+    document.getElementById("three").appendChild(back);
+}
+
+function keyDownEvents(rightWord, wordList) {
+    const buttons = document.querySelectorAll('button');
+    document.addEventListener('keydown', function (event) {
+        if (event.code === "Enter") {
+            buttons[buttons.length - 2].style.backgroundColor = "rgb(104, 157, 157)";
+            matchWord(rightWord, typedInput.join(""), wordList)
+            typedInput = [];
+            console.log(typedInput.join(""));
+        }
+        if (event.code === "Backspace") {
+            buttons[buttons.length - 1].style.backgroundColor = "rgb(104, 157, 157)";
+            erase();
+            typedInput.pop();
+            console.log(typedInput);
+        }
+        for (let i = 0; i < keyboardLetters.length; i++) {
+            if (event.code === `Key${alphabet[i]}`) {
+                buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor = "rgb(104, 157, 157)";
+                typedInput.push(alphabet[i]);
+                type();
+                console.log(typedInput);
+                console.log(alphabet[i]);
+            }
+        }
+    })
+    document.addEventListener('keyup', function (event) {
+        if (event.code === "Enter") {
```

Comment:
spacing, in general. do you use a code formatter like prettier? it can autoformat the code for you on save